src/Symfony/Component/Serializer/Normalizer/AbstractObjectNormalizer.php

```diff
@@ -132,6 +134,10 @@ public function normalize($object, $format = null, array $context = array())
             $data = $this->updateData($data, $attribute, $this->serializer->normalize($attributeValue, $format, $this->createChildContext($context, $attribute)), $class, $format, $context);
         }
 
+        if (isset($context[self::ARRAY_OBJECTS_KEY]) && !\count($data)) {
+            return new \ArrayObject();
```

Comment:
jms serializer is using `stdClass` for this. not sure which is better. what does json encode do with \ArrayObject?

Reply:
Its because the `Encoder`s will treat empty stdclass to `[]` in json - I'm using `ArrayObject` to explicitly say, this is an object, and it should stay an object.

Reply:
usually, json_encode makes `{}` out of stdclass: https://3v4l.org/pAZKW

(not saying its wrong to use ArrayObject, just trying to understand)

Reply:
It came from the original issue, https://github.com/symfony/symfony/issues/23019 - I think one big advantage is that its explicit, whereas stdclass should be normalized in most cases - i.e. when it has some keys.